hack/observability/capi-state-metrics/crd-config.yaml

```diff
@@ -0,0 +1,656 @@
+kind: CustomResourceStateMetrics
```

Comment:
Yeah, still working on it. At least to get https://github.com/kubernetes/kube-state-metrics/pull/1810 in which would allow us to split the file on a per CR base.

Reply:
sadly this pr did not make it as I thought. Still only a single file for now. Future kube-state-metrics (ksm) may support config from a ksm-specific CR instead of a static config file. But that needs to be implemented.

Fallback to single file for now.
I added a ugly shell script to generate it (`metrics/build.sh`), ok for now because this PR is WIP. Todo: find a solution for that. Kustomize is not able to help here :-(

Reply:
I think it's fine for the dev-env. Even splitting it in multiple files won't fix the big problem on how to keep that big of a YAML sync with our CRDs.

I think the eventual solution to this problem has to be a gen tool (but of course just a follow-up and in the "fullness of time" :))

Reply:
ack 👍